www/src/components/layer-model/layer-content-sections.js

```diff
@@ -0,0 +1,352 @@
+import React from "react"
+import { Link } from "gatsby"
+import MdLoop from "react-icons/lib/md/loop"
+
+import {
+  fontSizes,
+  space,
+  colors,
+  radii,
+  shadows,
+  transition,
+  mediaQueries,
+} from "../../utils/presets"
+
+// Components for building sections used in the model
+const LayerContentWrapper = ({ children }) => (
+  <div
+    css={{
+      padding: `${space[4]} 0`,
+      display: `grid`,
+      gridTemplateRows: `1fr 1fr`,
+      gridTemplateAreas: `"example" "content"`,
+      gridGap: 0,
+      [mediaQueries.md]: {
+        gridTemplateRows: `1fr`,
+        gridTemplateAreas: `"example content"`,
+        gridTemplateColumns: `1fr 1fr`,
+        gridGap: space[6],
+      },
+    }}
+  >
+    {children}
+  </div>
+)
+
+const ExampleWrapper = ({ children }) => (
+  <div
+    css={{
+      borderRadius: radii[3],
+      maxWidth: 620,
+      [mediaQueries.md]: {
+        maxWidth: 300,
+      },
+    }}
+  >
+    {children}
+  </div>
+)
+
+const pseudoProperties = {
+  boxShadow: shadows.floating,
+  color: colors.white,
+}
+// prettier-ignore
+const transitionProperty = `${transition.speed.default} ${transition.curve.default}`
+
+const CodeWrapper = ({
+  title,
+  language,
+  rotateButton = false,
+  sourceIndex,
+  setSourceIndex,
+  children,
+}) => (
+  <React.Fragment>
+    {title && (
+      <div
+        css={{
+          position: `relative`,
+          display: `flex`,
+          alignItems: `center`,
+          justifyContent: `space-between`,
+          borderTopRightRadius: radii[3],
+          borderTopLeftRadius: radii[3],
+        }}
+        className="gatsby-code-title"
+      >
+        <div css={{ fontSize: fontSizes[0] }}>{title}</div>
+        {rotateButton && (
+          <button
+            css={{
+              position: `absolute`,
+              right: space[3],
+              backgroundColor: `transparent`,
+              border: `none`,
+              color: colors.grey[60],
+              cursor: `pointer`,
+              padding: `${space[2]} ${space[2]}`,
+              transition: transitionProperty,
+              borderRadius: radii[2],
+              whiteSpace: `nowrap`,
+              ":focus": {
+                ...pseudoProperties,
+                backgroundColor: colors.purple[40],
+              },
+              ":hover": {
+                ...pseudoProperties,
+                backgroundColor: colors.purple[50],
+              },
+              ":active": {
+                ...pseudoProperties,
+                backgroundColor: colors.purple[60],
+              },
+              ":focus:before": {
+                content: `"cycle source "`,
+              },
+              ":hover::before": {
+                content: `"cycle source "`,
+              },
+            }}
+            onClick={() =>
+              setSourceIndex(
+                sourceIndex === sources.length - 1 ? 0 : sourceIndex + 1
```

Comment:
Writing this without testing, but neat little trick :)

```suggestion
                (sourceIndex + 1) % sources.length
```

it'll be the next index if not greater than the length of the array. Not necessary though, use your discretion if it's better!